util/topgen.py

```diff
@@ -1152,14 +1154,17 @@ def render_template(template_path: str, rendered_path: Path,
                         out_path / "rtl" / "autogen" / f"{top_name}.sv",
                         gencmd=gencmd)
 
-        # Multiple chip-levels (ASIC, FPGA, Verilator, etc)
-        for target in topcfg["targets"]:
-            target_name = target["name"]
-            render_template(top_template_path / "chiplevel.sv.tpl",
-                            out_path /
-                            f"rtl/autogen/chip_{topname}_{target_name}.sv",
-                            gencmd=gencmd,
-                            target=target)
+        # Render chip-level template only if top specifies one
+        chiplevel_tpl = top_template_path / 'chiplevel.sv.tpl'
+        if chiplevel_tpl.is_file():
```

Comment:
This seems to be inverted: I would expect topcfg["targets"] to be empty or not-such-entry if there is no chip-level.

Reply:
I dropped the commit. Let's leave `topcfg["targets"]` empty if you don't need an implementation target.